### PR TITLE
chore: intergrate communities

### PR DESCRIPTION
**Pull request by byronkimani**
- intergrate communities

### CODE REVIEW

File: lib/application/core/services/communities_utils.dart [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+  ).timeout(Duration(seconds: fullSync ? 180 : 60));
+
+  final dynamic data = await json.decode(utf8.decode(response.bodyBytes));
+
+  // Process the events here
+  final Map<String, dynamic> processedData = data as Map<String, dynamic>;
+
+  final SyncResponse syncResponse = SyncResponse.fromJson(processedData);
+
+  // TODO(abiud): process errors in the response. Attach to the backoff in state
```
Comment (abiudrn) [inline lib/application/core/services/communities_utils.dart:85]: Remove this TODO

File: lib/application/redux/actions/communities/communities_logout_action.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import 'package:pro_health_360/application/redux/states/app_state.dart';
+
+class CommunitiesLogoutAction extends ReduxAction<AppState> {
+  CommunitiesLogoutAction({this.onSuccess});
+
+  final Function()? onSuccess;
+
+  @override
+  void after() {
+    dispatch(WaitAction<AppState>.remove(signInFlag));
```
Comment (abiudrn) [inline lib/application/redux/actions/communities/communities_logout_action.dart:12]: Change this flag to `commSignOutFlag`

File: lib/application/redux/actions/communities/delete_message_action.dart [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        senderID: senderID,
+        eventID: eventID,
+        userID: userID,
+      );
+
+      final Map<String, dynamic> body =
+          json.decode(response.body) as Map<String, dynamic>;
+
+      if (response.statusCode == 403) {
+        // TODO(abiud): Check for perms error properly
```
Comment (abiudrn) [inline lib/application/redux/actions/communities/delete_message_action.dart:65]: Remove this TODO

File: lib/application/redux/flags/flags.dart [hunk 4]
```diff
@@ -139,3 +139,34 @@ const String fetchManagedClientsFlag = 'fetch_managed_clients_flag';
 const String setDefaultFacilityFlag = 'set_default_facility_flag';
 const String fetchProgramsFlag = '/fetchProgramsFlag';
 const String setCurrentProgramFlag = '/setCurrentProgramFlag';
+
+//communities
+const String signInFlag = 'signInFlag';
+
+const String signOutFlag = 'signOutFlag';
```
Comment (abiudrn) [inline lib/application/redux/flags/flags.dart:146]: These 2 flags might interfere with the normal platform flags, change them to have the `communities_*` prefix

File: lib/presentation/communities/invited_groups/pages/group_info_page.dart [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+          ),
+          const SizedBox(height: 20),
+          Align(
+            alignment: Alignment.centerRight,
+            child: TextButton(
+              key: inviteUsersKey,
+              onPressed: () async {
+                // Search for a user and add them
+                // navigate to rooms page
+                // ignore: use_build_context_synchronously
```
Comment (abiudrn) [inline lib/presentation/communities/invited_groups/pages/group_info_page.dart:111]: Remove this linting exception

File: lib/presentation/communities/invited_groups/pages/group_info_page.dart [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+                          ),
+                        );
+                      },
+                      client: AppWrapperBase.of(context)!.graphQLClient,
+                    ),
+                  );
+                },
+                child: const Padding(
+                  padding: EdgeInsets.all(8.0),
+                  child: Text('Leave room'),
```
Comment (abiudrn) [inline lib/presentation/communities/invited_groups/pages/group_info_page.dart:190]: Extract these strings into well named constants

File: lib/presentation/communities/invited_groups/pages/media_item_widget.dart [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+                      orElse: () {
+                        return null;
+                      },
+                    );
+
+                    if (imageByteMap?.isNotEmpty ?? false) {
+                      final Uint8List? bodyBytes = imageByteMap![widget.url];
+
+                      if (bodyBytes != null && bodyBytes.isNotEmpty) {
+                        // return Image(image: MemoryImage(bodyBytes));
```
Comment (abiudrn) [inline lib/presentation/communities/invited_groups/pages/media_item_widget.dart:123]: Remove dead code

File: lib/presentation/communities/invited_groups/pages/normal_message_item.dart [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+    return InkWell(
+      key: messageOptionsKey,
+      onTap: () {
+        // Open the message in a new page
+      },
+      onLongPress: () async {
+        await showModalBottomSheet(
+          context: context,
+          builder: (BuildContext context) {
+            // TODO!(abiud): populate these values dynamically based on power level
```
Comment (abiudrn) [inline lib/presentation/communities/invited_groups/pages/normal_message_item.dart:43]: Remove this TODO. It has already been handled in the implementation

File: lib/presentation/communities/invited_groups/pages/room_list_item_widget.dart [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+  final String groupName;
+  final Room currentRoom;
+  final bool isInvite;
+  final String groupMembers;
+
+  @override
+  Widget build(BuildContext context) {
+    return InkWell(
+      key: Key(groupName),
```
Comment (abiudrn) [inline lib/presentation/communities/invited_groups/pages/room_list_item_widget.dart:23]: Extract this key into the `app_widget_keys_file`

File: lib/presentation/communities/invited_groups/pages/room_list_item_widget.dart [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+                    fontWeight: FontWeight.w700,
+                    fontSize: 17,
+                  ),
+                ),
+                const SizedBox(height: 5),
+                Text(
+                  isInvite
+                      ? 'You were invited. Tap for more info'
+                      : '$groupMembers members',
+                  style: const TextStyle(
```
Comment (abiudrn) [inline lib/presentation/communities/invited_groups/pages/room_list_item_widget.dart:73]: Extract these strings. Do it for the whole folder

**Review by abiudrn — APPROVED** [review]
Approved. The strings will be extracted in a follow up MR

**Review by EugeneDevv — APPROVED** [review]

**Pull request merged**